velox/exec/fuzzer/ArbitrationFuzzer.cpp

```diff
@@ -0,0 +1,400 @@
+ public:
+  explicit ArbitrationFuzzer(
+      size_t initialSeed,
+      std::unique_ptr<ReferenceQueryRunner>);
+
+  void go();
+
+  struct PlanWithSplits {
+    core::PlanNodePtr plan;
+    std::vector<std::shared_ptr<connector::ConnectorSplit>> splits;
+
+    explicit PlanWithSplits(
+        core::PlanNodePtr _plan,
+        const std::vector<std::shared_ptr<connector::ConnectorSplit>>& _splits =
+            {})
+        : plan(std::move(_plan)), splits(_splits) {}
+  };
+
+ private:
+  static VectorFuzzer::Options getFuzzerOptions() {
+    VectorFuzzer::Options opts;
+    opts.vectorSize = FLAGS_batch_size;
+    opts.stringVariableLength = true;
+    opts.stringLength = 100;
+    opts.nullRatio = FLAGS_null_ratio;
+    return opts;
+  }
+
+  static inline const std::string kHiveConnectorId = "test-hive";
+  static inline const std::vector<std::string> operators = {
+      "OrderBy",
+      "HashJoin",
+      "TopN"};
+
+  // Makes a connector split from a file path on storage.
+  static std::shared_ptr<connector::ConnectorSplit> makeSplit(
+      const std::string& filePath);
+
+  void seed(size_t seed) {
+    currentSeed_ = seed;
+    vectorFuzzer_.reSeed(seed);
+    rng_.seed(currentSeed_);
+  }
+
+  void reSeed() {
+    seed(rng_());
+  }
+
+  void execute(core::PlanNodePtr plan, bool simple = false);
+
+  // Runs one test iteration from query plans generations, executions and result
+  // verifications.
+  void verify();
+
+  int32_t randInt(int32_t min, int32_t max) {
+    return boost::random::uniform_int_distribution<int32_t>(min, max)(rng_);
+  }
+
+  std::vector<RowVectorPtr> generateInput();
+  std::unordered_map<std::string, std::tuple<core::PlanNodePtr, RowVectorPtr>>
+  makeDefaultPlanResults(const std::vector<RowVectorPtr>& inputs);
+
+  FuzzerGenerator rng_;
+  size_t currentSeed_{0};
+  std::unordered_map<std::string, std::unordered_map<std::string, std::string>>
+      queryConfigs;
+
+  std::shared_ptr<memory::MemoryPool> rootPool_{
+      memory::memoryManager()->addRootPool(
+          "arbitrationFuzzer",
+          64L << 20,
+          memory::MemoryReclaimer::create())};
+  std::shared_ptr<memory::MemoryPool> pool_{rootPool_->addLeafChild(
+      "arbitrationFuzzerLeaf",
+      true,
+      exec::MemoryReclaimer::create())};
+  std::shared_ptr<memory::MemoryPool> writerPool_{rootPool_->addAggregateChild(
+      "arbitrationFuzzerWriter",
+      exec::MemoryReclaimer::create())};
+  VectorFuzzer vectorFuzzer_;
+  std::unique_ptr<ReferenceQueryRunner> referenceQueryRunner_;
+  std::shared_ptr<folly::Executor> executor_{
+      std::make_shared<folly::CPUThreadPoolExecutor>(
+          std::thread::hardware_concurrency())};
+};
+
+ArbitrationFuzzer::ArbitrationFuzzer(
+    size_t initialSeed,
+    std::unique_ptr<ReferenceQueryRunner> referenceQueryRunner)
+    : vectorFuzzer_{getFuzzerOptions(), pool_.get()},
+      referenceQueryRunner_{std::move(referenceQueryRunner)} {
+  filesystems::registerLocalFileSystem();
```

Comment:
Shall we just register these in velox/exec/fuzzer/ArbitrationFuzzerRunner.h? Thanks!